jwst/master_background/master_background_step.py

```diff
@@ -46,129 +47,164 @@ def process(self, input):
+                if self.save_background:
+                    self.save_model(master_background, suffix='masterbg', asn_id=asn_id)
             
             self.record_step_status(result, 'master_background', success=True)
 
         return result
 
 
+    @property
+    def _do_sub(self):
+        """
+        Decide if subtraction is to be done
+
+        Encapsulates logic that checks if background step has already been run
+        on the data, or if the user has selected to force_subtract regardless.
+
+        Returns
+        -------
+        do_sub : bool
+            If ``True``, do the subtraction
+        """
+        do_sub = True
+        input_data = self.input_data
+        if not self.force_subtract:
+
+            # check if the input data is a model container. If it is then loop over
+            # container and see if the background was subtracted in calspec2.
+            # If all data was background subtracted, skip master bgk subtraction.
+            # If there is a mixture of some being background subtracted, don't
+            # subtract and print warning message
+            if isinstance(input_data, datamodels.ModelContainer):
+                isub = 0
+                for indata in input_data:
+                    if indata.meta.cal_step.back_sub == 'COMPLETE':
+                        do_sub = False
+                        isub += 1
+
+                if not do_sub and isub == len(input_data):
+                    self.log.info(
+                        "Not subtracting master background, background was subtracted in calspec2")
+                    self.log.info("To force the master background to be subtracted from this data, "
+                        "run again and set force_subtract = True.")
+
+                if not do_sub and isub != len(input_data):
+                    self.log.warning("Not subtracting master background.")
+                    self.log.warning("Input data contains a mixture of data with and without "
+                        "background subtraction done in calspec2.")
+                    self.log.warning("To force the master background to be subtracted from this data, "
+                        "run again and set force_subtract = True.")
+            # input data is a single file
+            else:
+                if input_data.meta.cal_step.back_sub == 'COMPLETE':
+                    do_sub = False
+                    self.log.info(
+                        "Not subtracting master background, background was subtracted in calspec2")
+                    self.log.info("To force the master background to be subtracted from this data, "
+                        "run again and set force_subtract = True.")
+
+        return do_sub
+
+
+def copy_background_to_flux(spectrum):
+    """Copy the background column to the flux column in a MultiSpecModel"""
+    result = spectrum.copy()
+    for spec in result.spec:
+        spec.spec_table['FLUX'] = spec.spec_table['BACKGROUND']
```

Comment:
It looks to me as if you need a `.copy()` here, to avoid clobbering flux when zeroing out background.

Reply:
Thanks!  Fixed.